perun-core/src/main/java/cz/metacentrum/perun/core/impl/TasksManagerImpl.java

```diff
@@ -499,6 +499,20 @@ public Task getTask(int serviceId, int facilityId) {
 		}
 	}
 
+	@Override
+	public List<Task> listAllTasksForService(int serviceId) {
+		try {
+			// jdbc template cannot be null
+			return getMyJdbcTemplate().query(
+				"select " + taskMappingSelectQuery + ", " + FacilitiesManagerImpl.facilityMappingSelectQuery +
+					", " + ServicesManagerImpl.serviceMappingSelectQuery + " from tasks left join services on tasks.service_id = services.id " +
+					"left join facilities on facilities.id = tasks.facility_id where services.id = ?",
+				TASK_ROWMAPPER, serviceId);
+		} catch (EmptyResultDataAccessException ex) {
```

Comment:
I think this exception cannot be thrown by the query method. If no rows are found, it returns an empty list.

Reply:
Removed the try-catch block from this method. The same construction is used in other methods here, but I will leave that for possible cleanup later.